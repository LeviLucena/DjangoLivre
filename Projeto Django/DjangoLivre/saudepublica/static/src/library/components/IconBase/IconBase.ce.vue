<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
