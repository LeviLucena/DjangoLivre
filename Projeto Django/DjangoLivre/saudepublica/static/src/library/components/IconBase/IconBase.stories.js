import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
