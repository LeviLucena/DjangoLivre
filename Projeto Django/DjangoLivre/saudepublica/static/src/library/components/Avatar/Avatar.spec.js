import { shallowMount } from '@vue/test-utils'
import IconBase from '../IconBase/IconBase.ce.vue'
import BrAvatar from './Avatar.ce.vue'

describe('BrAvatar', () => {
  test('Deve verificar se o tipo do avatar é Letra.', () => {
    const textName = 'Jhon Doe'
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        name: textName,
      },
    })
    expect(wrapper.text()).toMatch('J')
  })

  test("Deve verificar se o tipo do avatar é iconico'.", () => {
    const typeAvatar = true
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        iconic: typeAvatar,
      },
      global: {
        stubs: {
          'icon-base': IconBase,
        },
      },
    })
    expect(wrapper.find('.fa-user').exists()).toBe(true)
  })

  test("Deve verificar se o tipo do avatar é fotográfico'.", () => {
    const urlImageAvatar = 'https://picsum.photos/id/823/400'
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        image: urlImageAvatar,
      },
    })
    expect(wrapper.find("img[src='" + urlImageAvatar + "']").exists()).toBe(true)
  })

  test('Deve verificar se a densidade aplica é small', () => {
    const densitySmall = 'small'
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        name: 'Jhon Doe',
        density: densitySmall,
      },
    })
    expect(wrapper.find('span.br-avatar.small').exists()).toBe(true)
  })

  test('Deve verificar se a densidade aplica é medium', () => {
    const densityMedium = 'medium'
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        name: 'Jhon Doe',
        density: densityMedium,
      },
    })
    expect(wrapper.find('span.br-avatar.medium').exists()).toBe(true)
  })

  test('Deve verificar se a densidade aplica é large', () => {
    const densitylarge = 'large'
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        name: 'Jhon Doe',
        density: densitylarge,
      },
    })
    expect(wrapper.find('span.br-avatar.large').exists()).toBe(true)
  })

  test('has the expected html structure', () => {
    const wrapper = shallowMount(BrAvatar, {
      propsData: {
        name: 'Jhon Doe',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
