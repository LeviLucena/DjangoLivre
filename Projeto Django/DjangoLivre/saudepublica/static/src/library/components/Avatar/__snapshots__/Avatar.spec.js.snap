// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrAvatar has the expected html structure 1`] = `
<span
  class="br-avatar small"
>
  <span
    class="image bg-support-01 text-secondary-01"
  >
    J
  </span>
</span>
`;
